api.js

```diff
@@ -426,6 +427,64 @@ API.prototype.noticeError = function noticeError(error, customAttributes) {
   this.agent.errors.addUserError(transaction, error, filteredAttributes)
 }
 
+/**
+ * Sends an application log message to New Relic. The agent already
+ * automatically does this for some instrumented logging libraries,
+ * but in case you are using another logging method that is not
+ * already instrumented by the agent, you can use this function
+ * instead.
+ *
+ * If application log forwarding is disabled in the agent
+ * configuration, this function does nothing.
+ *
+ * @param {object} logMessage  The log line object to send.
+ * @param {string} logMessage.message The log message.
+ * @param {string} logMessage.level The log level severity. If this key is
+ *   missing, it will default to UNKNOWN
+ * @param {number} logMessage.timestamp   Unix epoch number denoting the
+ *   time that this log message was produced. If this key is missing,
+ *   it will default to the output of `Date.now()`.
+ */
+API.prototype.recordLogEvent = function recordLogEvent(logMessage) {
+  const metric = this.agent.metrics.getOrCreateMetric(NAMES.SUPPORTABILITY.API + '/recordLogEvent')
+  metric.incrementCallCount()
+
+  if (!applicationLogging.isLogForwardingEnabled(this.agent.config, this.agent)) {
+    logger.warnOnce(
+      'Record logs',
+      'Application log forwarding disabled, method API#recordLogEvent will not record messages'
+    )
+    return
+  }
+
+  // If they don't pass a logMessage object, or it doesn't have the
+  // required `message` key, bail out.
+  if (!logMessage || typeof logMessage !== 'object' || logMessage.message === undefined) {
+    logger.warn(
+      'recordLogEvent requires an object with a `message` attribute for its single argument, got %s (%s)',
+      stringify(logMessage),
+      typeof logMessage
+    )
+    return
+  }
+
+  if (logMessage.level === undefined) {
```

Comment:
if we are doing all of this checking. we may want to consider the renaming of the error attributes and truncating.  You can see it being done here:

 * [pino](https://github.com/newrelic/node-newrelic/blob/main/lib/instrumentation/pino/pino.js#L129)
 * [winston](https://github.com/newrelic/node-newrelic/blob/main/lib/instrumentation/nr-winston-transport.js#L77)
 * [bunyan](https://github.com/newrelic/node-newrelic/blob/main/lib/instrumentation/bunyan.js#L39)